### PR TITLE
bring in rackup for rack v3 compatibility

### PR DESCRIPTION
**Pull request by fallwith**
With Rack v3, `rack/handler` has been moved out into its own new
`rackup` gem.

Conditionally bring in `rackup` only when Rack v3+ is in play.

Dynamically have the fake server class (CI) make use of `rackup` if it exists.

Remove the Mongrel RPM script, as Mongrel no longer appears to be
  supported via `rackup`.

Jump through some hoops for Ruby version and gem version combinations for instrumented gems.

### CODE REVIEW

File: test/multiverse/suites/excon/Envfile [hunk 1]
```diff
@@ -9,13 +9,25 @@ excon_versions = [
   '0.64.0',
   '0.70.0',
   '0.78.1',
-  '0.85.0',
+  '0.85.0'
 ]
 
 def gem_list(excon_version = nil)
+  # TODO: CI - rack v3.0.0 causes 2 test failures
```
Comment (tannalynn) [inline test/multiverse/suites/excon/Envfile:16]: Could you make an issue for us to address all these failures mentioned in the todos to make sure we don't lose track of it?

**Review by tannalynn — APPROVED** [review]
Thank you for working on this! A lot was involved!

File: test/multiverse/suites/excon/Envfile (hunk 1 above)
Comment (tannalynn) [reply test/multiverse/suites/excon/Envfile:16]: it looks like you did that at the same time as I submitted the comment lol. thnak you!

**Comment by fallwith** [comment]
Issue #1393 created for the TODOs for all of the HTTP clients
Issue #1394 crested for the TODO for Unicorn

File: test/multiverse/suites/excon/Envfile (hunk 1 above)
Comment (fallwith) [reply test/multiverse/suites/excon/Envfile:16]: Done, that particular one is covered by #1393 which seems to impact all of the HTTP clients with new TODO statements, and the separate #1394 will track the new TODO for Unicorn.

File: test/environments/norails/Gemfile [hunk 2]
```diff
@@ -6,7 +6,7 @@ gem 'rake', '~> 12.3.3'
 gem 'minitest', '5.3.3'
 gem 'minitest-stub-const', '~> 0.6'
 gem 'mocha', '~> 1.9.0', :require => false
-gem 'rack', '>= 2.1.4'
+gem 'rack', "~> #{RUBY_VERSION >= '2.3.0' ? '2.2.4' : '2.1.4.1'}"
```
Comment (kaylareopelle) [inline test/environments/norails/Gemfile:9]: Does this prevent us from testing Rack 3 in `norails`?

File: test/environments/rails50/Gemfile [hunk 3]
```diff
@@ -8,7 +8,7 @@ gem 'rails', '~> 5.0.0'
 gem 'minitest', '5.2.3'
 gem 'minitest-stub-const', '~> 0.6'
 gem 'mocha', '~> 1.1.0', :require => false
-gem 'rack', '< 2.1.0'
+gem 'rack', "~> #{RUBY_VERSION >= '2.3.0' ? '2.2.4' : '2.1.4.1'}"
```
Comment (kaylareopelle) [inline test/environments/rails50/Gemfile:11]: This is only tested on JRuby at this time, so the condition may not be necessary

**Review by kaylareopelle — APPROVED** [review]
I like that you added the text for the errors to each todo in the Envfiles! One small question, otherwise LGTM!

File: test/environments/norails/Gemfile (hunk 2 above)
Comment (fallwith) [reply test/environments/norails/Gemfile:9]: This change was intended to just be more specific with the Ruby 2.2 and Ruby 2.3+ specific versions of Rack 2. It's the previous existence of the `rack-test` constraint to a super old version (`< 0.8.0`) which was already keeping us on Rack 2 / `< Rack 3`.

File: test/environments/rails50/Gemfile (hunk 3 above)
Comment (fallwith) [reply test/environments/rails50/Gemfile:11]: Ah okay, cool. I was just drawing a line between Rails 7/edge and all others and giving all the others this same treatment.

**Pull request merged**